engine/src/main/java/org/camunda/bpm/engine/impl/cmd/DeleteProcessInstanceCommentCmd.java

```diff
@@ -0,0 +1,118 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl.cmd;
+
+import static org.camunda.bpm.engine.impl.util.EnsureUtil.ensureNotNull;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import org.camunda.bpm.engine.ProcessEngineException;
+import org.camunda.bpm.engine.history.UserOperationLogEntry;
+import org.camunda.bpm.engine.impl.cfg.CommandChecker;
+import org.camunda.bpm.engine.impl.interceptor.Command;
+import org.camunda.bpm.engine.impl.interceptor.CommandContext;
+import org.camunda.bpm.engine.impl.persistence.entity.CommentEntity;
+import org.camunda.bpm.engine.impl.persistence.entity.ExecutionEntity;
+import org.camunda.bpm.engine.impl.persistence.entity.PropertyChange;
+import org.camunda.bpm.engine.impl.persistence.entity.TaskEntity;
+import org.camunda.bpm.engine.task.Comment;
+
+/**
+ * Command to delete a comment by a given commentId and processInstanceId or to delete all comments
+ * of a given processInstanceId
+ */
+
+public class DeleteProcessInstanceCommentCmd implements Command<Object>, Serializable {
+
+  private static final long serialVersionUID = 1L;
+  protected String commentId;
+  protected String processInstanceId;
+
+  public DeleteProcessInstanceCommentCmd(String processInstanceId, String commentId) {
+    this.processInstanceId = processInstanceId;
+    this.commentId = commentId;
+  }
+
+  public DeleteProcessInstanceCommentCmd(String processInstanceId) {
+    this.processInstanceId = processInstanceId;
+  }
+
+  public Object execute(CommandContext commandContext) {
+    if (processInstanceId == null && commentId == null) {
+      throw new ProcessEngineException("Both process instance and comment ids are null");
+    }
+
+    ensureNotNull("processInstanceId", processInstanceId);
+
+    if (commentId != null && processInstanceId != null) {
+      CommentEntity comment = commandContext.getCommentManager()
+          .findCommentByProcessInstanceIdAndCommentId(processInstanceId, commentId);
+      if (comment != null) {
+
+        TaskEntity task = getTask(comment, commandContext);
+
+        checkTaskWork(task, commandContext);
+        commandContext.getDbEntityManager().delete(comment);
+        logOperation(comment, task, null, commandContext);
+        task.triggerUpdateEvent();
+      }
+    } else {
+      ExecutionEntity processInstance = commandContext.getExecutionManager().findExecutionById(processInstanceId);
+      ensureNotNull("No processInstance exists with processInstanceId: " + processInstanceId, "processInstance: ",
+          processInstance);
+
+      List<Comment> comments = commandContext.getCommentManager().findCommentsByProcessInstanceId(processInstanceId);
+      if (!comments.isEmpty()) {
+        TaskEntity task = commandContext.getTaskManager().findTaskById(comments.get(0).getTaskId());
```

Comment:
📓 No action required, just note. This looks strange but so far we had only `#checkTaskWork()` by task so it's fine as we don't want to go the overkill to introduce check by process instance 🤷 .